### PR TITLE
Change the WebDriver client to provide a known type for window/new

### PR DESCRIPTION
**Pull request by gsnedders**
See also: https://github.com/w3c/webdriver/issues/1551. At the moment
the spec implies that any type is allowed, but implementations largely
enforce restrictions similar to elsewhere in the spec. Notably, Safari
returns error with error code invalid argument when the type is null.

This ensures the default is something that's well-defined, and allows
for potential edge-cases such as null to be tested separately. Tests for
these edge-cases should be written after the spec issue is resolved.

### CODE REVIEW

File: tools/webdriver/webdriver/client.py [hunk 1]
```diff
@@ -581,7 +581,7 @@ def source(self):
         return self.send_session_command("GET", "source")
 
     @command
-    def new_window(self, type_hint=None):
+    def new_window(self, type_hint="window"):
```
Comment (whimboo) [inline tools/webdriver/webdriver/client.py:584]: Can we just use `tab` here? It's less resource intensive than windows and all browsers actually support it.

**Review by whimboo — APPROVED** [review]
Fine for me. But please get another one to review. Maybe @jgraham?

**Review by stephenmcgruer — APPROVED** [review]
I think this is fine. At first I was concerned about `webdriver/tests/new_window/*`, but it seems they (sensibly) do not use this method but instead define their own that sends session commands directly (so they know they aren't getting any clever default behavior like this method produces ;)).

Looks like `webdriver/tests/new_window/new.py::test_type_with_null_value` should test the null behavior?

**Comment by gsnedders** [comment]
> Looks like webdriver/tests/new_window/new.py::test_type_with_null_value should test the null behavior?

see:

> Tests for these edge-cases should be written after the spec issue is resolved.

**Comment by stephenmcgruer** [comment]
That is what I was (implicitly, I admit) replying to - those tests appear to already exist (null in that particular existing case, other invalid types elsewhere in that same test file). Perhaps poor linguistics on my part with 'should'; a more correct statement might have been: 'Looks like webdriver/tests/new_window/new.py::test_type_with_null_value does test the null behavior?'

**Pull request merged**